drivers/interrupt_controller/intc_plic.c

```diff
@@ -152,12 +238,15 @@ int riscv_plic_get_irq(void)
 {
 	return save_irq;
 }
-
-static void plic_irq_handler(const void *arg)
+const struct device *riscv_plic_get_dev(void)
 {
-	volatile struct plic_regs_t *regs =
-	    (volatile struct plic_regs_t *) PLIC_REG;
+	return save_dev;
```

Comment:
This will get very messy, very fast.

Reply:
`save_dev` is the interrupt controller of the IRQ(`save_irq`) that's active